File: hw/chicaiza/u3/HW22Strategy/sortapp/src/main/java/ec/edu/espe/sort/model/BubbleSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.sort.model;

/**
 *
 * @author Chicaza Michael
 */
public class BubbleSort implements SortingStrategy{
    public int[] sort(int data[]){
        int n = data.length;
        int[] result = data.clone();

        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (result[j] > result[j + 1]) {
                    int temp = result[j];
                    result[j] = result[j + 1];
                    result[j + 1] = temp;
                }
            }
        }

        System.out.println("Using BubbleSort");
        return result;
        
    }
    
    
}

File: hw/fuentes/U3/HW22STRATEGY/ExamU3Team3/SortAppPY/ec/edu/espe/clientApp/model/bubble_sort.py
from model.sorting_strategy import SortingStrategy

class BubbleSort(SortingStrategy):
    def sort(self, data: list[int]) -> list[int]:
        result = data.copy()
        n = len(result)
        for i in range(n):
            for j in range(0, n - i - 1):
                if result[j] > result[j + 1]:
                    result[j], result[j + 1] = result[j + 1], result[j]
        print("Using BubbleSort")
        return result
